tree.go

```diff
@@ -10,8 +10,16 @@ import (
 // TreeConstructorFn creates a fresh Tree instance to be used as the Merkle inside of rsmt2d.
 type TreeConstructorFn = func() Tree
 
+// CellIndex contains all information needed to identify the cell that is being
+// pushed
+type CellIndex struct {
+	AxisIndex uint
+	CellIndex uint
```

Comment:
Is it on purpose that the type CellIndex contains a field with the same name as the type (CellIndex)?

Reply:
whoops, changed the type `CellIndex` -> `SquareIndex`, and dropped the `Index` of the fields. Happy to change it something else if anyone has some ideas https://github.com/lazyledger/rsmt2d/pull/20/commits/1f35c00fc7dbd9b0e5dbecb4001d9acefdf9f18c